code/ui/ui_main.c

```diff
@@ -7450,24 +7480,28 @@ UI_MouseEvent
 void _UI_MouseEvent( int dx, int dy )
 {
 	// update mouse screen position
-	uiInfo.uiDC.cursorx += dx * uiInfo.screenXFactor;
+	if ( (ui_widescreenCursorScale.integer & 1) )
+		uiInfo.uiDC.cursorx += dx * uiInfo.screenXFactor;
+	else
+		uiInfo.uiDC.cursorx += dx;
+
 	if (uiInfo.uiDC.cursorx < 0)
 		uiInfo.uiDC.cursorx = 0;
 	else if (uiInfo.uiDC.cursorx > SCREEN_WIDTH)
 		uiInfo.uiDC.cursorx = SCREEN_WIDTH;
 
-	uiInfo.uiDC.cursory += dy;
+	if (ui_widescreen.integer && (ui_widescreenCursorScale.integer & 2) && uiInfo.portraitMode && isMainMenu)
+		uiInfo.uiDC.cursory += dy * uiInfo.screenYFactor;
```

Comment:
Always multiply by uiInfo.screenYFactor just set it accordingly to current portraitMode/mainMenu state